spring-cloud-gcp-autoconfigure/src/main/java/com/google/cloud/spring/autoconfigure/firestore/GcpFirestoreAutoConfiguration.java

```diff
@@ -150,12 +160,27 @@ public FirestoreTemplate firestoreTemplate(
           firestoreMappingContext);
     }
 
+    @Bean
+    @ConditionalOnMissingBean(name = "firestoreRoutingHeadersInterceptor")
+    public ClientInterceptor firestoreRoutingHeadersInterceptor() {
+      // add routing header for custom database id
+      Metadata routingHeader = new Metadata();
+      Metadata.Key<String> key =
+          Metadata.Key.of(Headers.DYNAMIC_ROUTING_HEADER_KEY, Metadata.ASCII_STRING_MARSHALLER);
+      routingHeader.put(key,
```

Comment:
This may work for this specific case, but routing header config could change and I'm not sure it works for all the new [routing header configs](https://github.com/googleapis/java-firestore/pull/1418/files#diff-8c7c9e5aae7297f519f9531432656cb5c8eaa1548796e5c60a091252a3cbc24eR301-R463), or maybe those are not required. 
In long term, switching to GAPIC stubs should solve this issue, but I'm not sure it is compatible with the Reactive paradigm. Why did we choose to use the gRPC generated stubs instead of GAPIC ones? Only because of Reactive?

Reply:
I see. My concern is that every time the backend makes the routing header required for a new RPC, or change how the routing header is constructed, we would have to come back and manually change it in spring-cloud-gcp. It may not in scope of this PR, can we investigate the possibility of switching to the GAPIC stubs?